src/huggingface_hub/validation.py

```diff
@@ -0,0 +1,193 @@
+from typing import Dict, List, Optional, Union
```

Comment:
If possible we should use the one already defined in `api-inference-community` otherwise we're exposed to desyncing both.

Reply:
I agree. I just wanted to avoid the dependency upon `api-inference-community` which also brings starlette and other dependencies that we might not want to have.

I think that the types/validation should go in another place tbh (there's already a TODO few lines below). Happy to depend on `api-inference-community` for the time being though

Reply:
Understandable, but there's also transformers/pipelines to take into account. (that's currently only loosely coupled with `api-inference-community`)

Reply:
oh yeah `pip install huggingface_hub` should only bring no to minimal number of dependencies

Reply:
(it's pulled by transformers, datasets, etc)

Reply:
Obsolete now :)